### PR TITLE
Upgrade to django-filter v2

### LINKED ISSUE #136: Upgrade to django-filter v2

### PR DESCRIPTION
**Pull request by DavisRayM**
- [x] Remove Backports Library
- [x] Drop Python2 and Python3.5 Support
- [x] Update Tests
- [x] Update Modules ( Models, Serializers, Viewsets, etc.)

Fix #136 

Previous PR #138

### CODE REVIEW

File: tasking/apps.py [hunk 1]
```diff
@@ -20,8 +21,6 @@ def ready(self):
         Do stuff when the app is ready
         """
         # set up app settings
-        from django.conf import settings
```
Comment (moshthepitt) [inline tasking/apps.py]: I think that it is not safe to move these to the top, as you want the code to run when this app has been registered by Django and not before.  This gives a hint about this idea https://docs.djangoproject.com/en/3.0/ref/applications/#django.apps.AppConfig.ready - but feel free to investigate furtherh.

File: tasking/filters.py [hunk 2]
```diff
@@ -1,10 +1,8 @@
-# -*- coding: utf-8 -*-
```
Comment (moshthepitt) [inline tasking/filters.py]: Could you comment on what necessitated the changes in this module?

File: tests/models/test_submissions.py [hunk 3]
```diff
@@ -25,7 +21,7 @@ def test_submission_model_str(self):
         submission = mommy.make(
             'tasking.Submission',
             task=cattle,
-            _fill_optional=['user', 'comment', 'submission_time'])
```
Comment (moshthepitt) [inline tests/models/test_submissions.py]: What are the effects of changing from `comment` to `comments`?

File: tests/models/test_tasks.py [hunk 4]
```diff
@@ -20,8 +16,8 @@ def test_task_model_str(self):
         Test the str method on Task model
         """
         cow_price = mommy.make('tasking.Task', name="Cow prices")
-        expected = 'Cow prices - {}'.format(cow_price.pk)
-        self.assertEqual(expected, six.text_type(cow_price))
+        expected = f'Cow prices - {cow_price.pk}'
+        self.assertEqual(expected, str(cow_price))
```
Comment (moshthepitt) [inline tests/models/test_tasks.py]: is the `str` necessary?

File: tasking/filters.py (hunk 2 above)
Comment (DavisRayM) [reply tasking/filters.py]: Initially thought this was a python2 feature.

Comment (DavisRayM) [reply tasking/filters.py]: The new `django-filter` doesn't support passing lists to `lookup_expr` like we used to do before. Needed to create a separate filter backend `TaskOccurrenceFilter` to handle the filtering of task occurrences from the tasks endpoint.

File: tests/models/test_submissions.py (hunk 3 above)
Comment (DavisRayM) [reply tests/models/test_submissions.py]: The `comment` field doesn't exist for the Submission model.

File: tests/models/test_tasks.py (hunk 4 above)
Comment (DavisRayM) [reply tests/models/test_tasks.py]: This test was testing the string representation of the object. I can replace this with a call to `__str__()` though.

File: tasking/apps.py (hunk 1 above)
Comment (DavisRayM) [reply tasking/apps.py]: Read more on this. Totally agree with moving the functions into the AppConfig. I'll revert this change.

Comment (DavisRayM) [reply tasking/apps.py]: Reverted the change.

**Pull request merged**